bleak/backends/scanner.py

```diff
@@ -142,6 +142,11 @@ def __init__(
 
         self.seen_devices = {}
 
+    @property
+    def detection_callback(self) -> Optional[AdvertisementDataCallback]:
+        """Get the callback provided to :meth:`__init__` or registered with :meth:`register_detection_callback`"""
+        return self._callback
+
```

Comment:
Instead of adding this property and doing the check for existing callback in the new `devices()` method, could we just add the check for `if self._callback` in `register_detection_callback()`?

Reply:
`BaseBleakScanner::register_detection_callback()` is not aware whether it is being called with user callback from `BleakScanner::__init__()` or `BleakScanner::devices()`, so it cannot perform any checks. Also the `register_detection_callback()`'s docstring says
> If another callback has already been registered, it will be replaced with ``callback``. ``None`` can be used to remove the current callback.

I cannot just overwrite the callback the user provided in `BleakScanner::__init__()` without their knowledge. So some check needs to be somewhere.
This is why the method of adding `queue` to `BaseBleakScanner` - in this case, any user callback and async iterator operation can work simultaneously, not exlusivelly.

Reply:
Thanks for you thoughts on this. It has given me an idea to perhaps change things in the backends to allow multiple callbacks to be registered at the same time. We don't have to do this as part of this pull request. So for now, how about we either silently overwrite the callback or just add a check for `if self._backend._callback` instead of adding this property?

Reply:
Done! In the meantime, I also pushed one example where both methods could be used simultaneously  https://github.com/hbldh/bleak/pull/1361#discussion_r1267361215